the question i wanted to find an answer to is:
What organizations are mentioned most often in the genAI for coding public discourse?

the steps i'm following are:
finding a list of names of companies that are related to AI and check how many of the total tweets are related to them

at first i decided to find the unique values of each of the columns and save those in files named after the columns to check if that would give me any insight

In [1]:
import pandas as pd
import os
from os import path
import json
import seaborn
import matplotlib as plt
import numpy as np
import math
import requests

In [2]:
import math

In [3]:
dfPath = path.join(os.getcwd(), "..", "ChatGPT_Tweets_Dataset_Full.csv")
df = pd.read_csv(dfPath)

In [4]:
# Extract unique values from each column
unique_values = {}
for column in df.columns:
    unique_values[column] = df[column].unique()

In [5]:
del unique_values["Unnamed: 0"]
del unique_values["tweet_id"]
del unique_values["original_text"]

In [6]:
newUniqueValues = {}
for column, values in unique_values.items():
    newUniqueValues[column] = []
    for value in values:
        if isinstance(value, str):   
            newUniqueValues[column].extend(list(set(value.split(", "))))
    newUniqueValues[column] = list(set(newUniqueValues[column]))

In [7]:
for column, values in newUniqueValues.items():
    with open(path.join(os.getcwd(), f"{column}_unique_values.txt"), "w", encoding="utf-8") as file:
        for value in values:
            file.write(str(value) + "\n")

after that i went searched datasets on keaggle to get a list of names of ai companies
the query i used was "ai company"

In [8]:
companiesNames = []


the first one i found is this https://www.kaggle.com/datasets/khaiid/startups-by-valuation
which has 5 columns: company, valuation, valuation_date, industry and country
the datasets results interesting since we can filter the companies by industry (still have to choose which ones to use)

i would like to specify that, even tho the name of the file is "startups.csv" the companies inside are both startups and non (i.e.: the first two of the datasets are bytedance and spacex)

In [9]:
startUpsDf = pd.read_csv ("Startups.csv")

In [10]:
startUpsDf.shape

(250, 5)

as we can see the total amount of companies is 250, which is not really enough for out research so i decided to look for new datasets

but before that some more data exploration

In [11]:
startUpsDf.columns

Index(['Company', 'Valuation', 'Valuation_date', 'Industry', 'Country'], dtype='object')

In [12]:
startUpsDf["Industry"].unique()

array(['Internet', 'Aerospace', 'Financial services', 'Fintech',
       'Graphic design', 'Retail', 'Software', 'Video games',
       'Cryptocurrency', 'Education technology', 'E-commerce',
       'Technology', 'Financial Technology', 'Finance', 'Health', nan,
       'Artificial intelligence', 'Batteries', 'Self-driving cars',
       'Marketplace', 'Venture capital', 'Internet media',
       'Computer software', 'Software as a service', 'Hospitality',
       'Marketing', 'Software development', 'Robotics',
       'Application security', 'Cloud security', 'Fantasy sports',
       'Logistics', 'Virtual events', 'NFT', 'Computer security',
       'Transportation', 'Travel', 'Robotic process automation',
       'Healthcare', 'Real estate technology', 'Workforce management',
       'Real estate', 'Cybersecurity', 'Collaborative software',
       'Mass media', 'Data infrastructure', 'Health technology',
       'Food delivery', 'Data governance', 'Ghost kitchen',
       'Electronic Cigarettes

we could filter the companies by the industry (i.e.: using only the companies that stricly related to it such as: Artificial intelligence, Data analytics, Education Technology, Educational technology) but that would have meant excluding the companies that get indirectly effected by gen ai (pejorative)

other than we would have a very small dataset

In [13]:
columnsToConsider = ["Artificial intelligence", "Data analytics", "Education Technology", "Educational technology"]
tempStartUpsDf = startUpsDf[startUpsDf["Industry"].isin(columnsToConsider)]
tempStartUpsDf.shape

(13, 5)

In [14]:
companiesNames.extend(list(set(startUpsDf["Company"])))

the second dataset i found is https://www.kaggle.com/datasets/iamsouravbanerjee/software-professional-salaries-2022

In [15]:
softwareProfessionalSalaries = pd.read_csv("Software_Professional_Salaries.csv")

In [16]:
softwareProfessionalSalaries.shape

(22774, 6)

this dataset is, in general, remarkably bigger than the other one, but we still have to filter by unique values of companies 

In [17]:
softwareProfessionalSalaries.columns

Index(['Rating', 'Company Name', 'Job Title', 'Salary', 'Salaries Reported',
       'Location'],
      dtype='object')

In [18]:
softwareProfessionalSalariescompanyNames = list(set(list(softwareProfessionalSalaries["Company Name"])))
len (softwareProfessionalSalariescompanyNames)

11263

we have many more names for sure

In [19]:
companiesNames.extend(softwareProfessionalSalariescompanyNames)

In [20]:
len(companiesNames)

11513

the next dataset i found is https://www.kaggle.com/datasets/joyshil0599/data-science-jobs-comprehensive-dataset

In [21]:
dataScienceJobsDf = pd.read_csv("data_science_job.csv", encoding="latin1")

In [22]:
dataScienceJobsDf.shape

(3198, 8)

In [23]:
dataScienceJobsDf.columns

Index(['Company', 'Job Title', 'Location', 'Job Type', 'Experience level',
       'Salary', 'Requirment of the company ', 'Facilities'],
      dtype='object')

In [24]:
dataScienceJobsDfCompaniesList = list(set(list(dataScienceJobsDf["Company"])))
len(dataScienceJobsDfCompaniesList)

1107

In [25]:
companiesNames.extend(dataScienceJobsDfCompaniesList)

In [26]:
len(companiesNames)

12620

and remove reduncancy

In [27]:
companiesNames = list(set(companiesNames))

In [28]:
len(companiesNames)

12471

In [29]:
companiesNames

['VRIZE',
 nan,
 'OpCord',
 'IT Division',
 'Gameskraft',
 'WTA Studios',
 'NeoSOFT',
 'NCIS',
 'O&P',
 'Boston Consulting Group',
 'TecOrb Technologies',
 'Keysight Technologies',
 'Infiniti Research',
 'RDA Labs',
 'Tracxn',
 'CYBERNETYX Interaktiv',
 'Maxim Information Technologies',
 'Animaker',
 'Mintware Platforms',
 'Kenneth David & Co.',
 'VAYUZ Technologies',
 'Healofy',
 'Infygen Technologies',
 'Elvya Technologies',
 'Sequoia',
 'California Software',
 'entomo',
 'Mindrops',
 'eRevMax',
 'BPK Technology',
 'Scopely',
 'Licious',
 'Logicshore Services',
 'Viva Wallet',
 'Mindtree',
 'Swar Yantra Technologies',
 'Winklix Internet',
 'Ideal IT',
 'UrbanKisaan',
 'INCITE',
 'ExpansionJS',
 'FITSO',
 'TRT',
 'jiko',
 'DKG Labs',
 'Insight Financial Solutions',
 'Tekvisions',
 'Albuquerque Isotopes',
 'Consult IT',
 'DXC Technology',
 'Introtuce',
 'Arjak Mondal',
 'GEP',
 'Yepme.com',
 'Eloit',
 'SonicWall',
 'Luxoft',
 'I Have A Future',
 'Stonedge Technologies and Robotics',
 '

than i switched to another query on keggle, using just the word "company"
i found a dataset which is 2.08 gbs with a total of 16695992 unique company names
i decided to finish my search with that one

https://www.kaggle.com/datasets/mfrye0/bigpicture-company-dataset

at first tho i needed to save the list i created so far to file and restart jupyter, otherwise it would be too heavy for my laptop

In [30]:
import pickle 

In [31]:
with open ("companiesNames.pickle", "wb") as f:
    pickle.dump(companiesNames, f)

In [32]:
companiesNames = []
with open ("companiesNames.pickle", "rb") as f:
    companiesNames = pickle.load(f)

In [33]:
len (companiesNames)

12471

In [34]:
companies2023Q4SmDf = pd.read_csv("companies-2023-q4-sm.csv")

this dataset is not on github since it's too heavy

In [35]:
companies2023Q4SmDf.shape

(17154017, 10)

since the number of rows is too big i decided to filter them out using the size of the company

as reported on the keaggle page the size indicates the number of people in the company and 49% of the companies have just 1 to 10 employees, another 22% only has 11 to 50 and the remaining 29% has more

if we are going to find references in the tweets they probably are going to be, most likely, from medium-big companies
i decided than to remove the companies that have less than 51 employees

In [36]:
companies2023Q4SmDf.columns

Index(['handle', 'name', 'website', 'industry', 'size', 'type', 'founded',
       'city', 'state', 'country_code'],
      dtype='object')

In [37]:
filtered_df = companies2023Q4SmDf[~companies2023Q4SmDf['size'].isin(['1-10', '11-50'])]

In [38]:
filtered_df.shape

(4962543, 10)

since we still have too many values i decided to filter away the ones that dont have a website

In [39]:
df_filtered2 = filtered_df[filtered_df['website'].notnull()]

In [40]:
df_filtered2.shape

(3665161, 10)

if a company has null in the founded column it probably doesnt have much relevance

In [41]:
df_filtered3 = df_filtered2[df_filtered2['founded'].notnull()]

In [42]:
df_filtered3.shape

(1293668, 10)

than i checked the type column and there are a couple of values that make the company probably irrelevant

In [43]:
df_filtered3["type"].unique()

array(['Privately Held', 'Public Company', 'Partnership',
       'Government Agency', 'Self-Owned', 'Nonprofit', 'Self-Employed',
       'Educational', nan], dtype=object)

In [44]:
df_filtered4 = df_filtered3[~df_filtered3['type'].isin(['Self-Owned', 'Self-Employed'])]

In [45]:
df_filtered4.shape

(1198399, 10)

since there were still too many couples i decided to filter away the ones that dont have a value in any of the columns

In [46]:
df_filtered5 = df_filtered4.dropna()

In [47]:
df_filtered5.shape

(912899, 10)

we got to basially 1 million companies

since this i decided to keep all the remaining companies

now we add all the companies to the list and save them on file

In [48]:
df_filtered5.columns

Index(['handle', 'name', 'website', 'industry', 'size', 'type', 'founded',
       'city', 'state', 'country_code'],
      dtype='object')

In [49]:
companiesNames.extend(list(set(list(df_filtered5["name"]))))

In [50]:
len (companiesNames)

917331

In [51]:
with open ("companiesNames.pickle", "wb") as f:
    pickle.dump(companiesNames, f)